coverage.gradle

```diff
@@ -0,0 +1,97 @@
+def excludes = [
+        // DI graph
+        '**/*Module.*',
+        '**/*Module*.*',
+
+        // Android composables
+        '**/*Screen*',
+        '**/components/*',
+        '**/*Compose*.*',
+
+        // Android framework
+        '**/*Activity*',
+        '**/*AndroidService*',
+        '**/*Application*',
+
+        // Generated
+        '**/*serializer*',
+        '**/*Serializer*',
+        "**/*request/*Companion*.*",
+        '**/*QueriesImpl*',
+        '**/*Db*',
+        '**/Select*',
+
+        // Tmp until serializationx can ignore generated
+        '**/Api*',
+]
+
+def initializeReport(report, projects, classExcludes) {
+    projects.each { project -> project.apply plugin: 'jacoco' }
+    report.executionData { fileTree(rootProject.rootDir.absolutePath).include("**/build/jacoco/*.exec") }
+
+    report.reports {
+        xml.enabled true
+        html.enabled true
+        csv.enabled false
+    }
+
+    gradle.projectsEvaluated {
+        def androidSourceDirs = []
+        def androidClassDirs = []
+
+        projects.each { project ->
+            switch (project) {
+                case { project.plugins.hasPlugin("com.android.application") }:
+                    androidClassDirs.add("${project.buildDir}/tmp/kotlin-classes/debug")
+                    androidSourceDirs.add("${project.projectDir}/src/main/kotlin")
+                    androidSourceDirs.add("${project.projectDir}/src/main/java")
+                    break
+                case { project.plugins.hasPlugin("com.android.library") }:
+                    androidClassDirs.add("${project.buildDir}/tmp/kotlin-classes/release")
```

Comment:
we can replace  `kotlin-classes/release` with `kotlin-classes/debug` as the debug tasks (with the current project setup) don't compile library module release classes